scripts/publish.sh

```diff
@@ -15,12 +15,18 @@ if [[ -z $PUBLISH_NPM_TAG ]]; then
   exit 1
 fi
 
+CURRENT_PACKAGE_VERSION=$(jq --raw-output .version package.json)
+
+if [[ "$CURRENT_PACKAGE_VERSION" = "0.0.0" ]]; then
+  echo "Notice: Invalid version: $CURRENT_PACKAGE_VERSION. aborting publish."
+  exit 0
```

Comment:
Nit: the checks above are not consistent, but wouldn't make more sense to exit with a non-zero code? This way the CI will visibly fail

```suggestion
  exit 1
```

Reply:
I believe we had the same conversation about the check below this, for when the package is already published.

Do we really want to fail CI if one of the packages is 0.0.0? This script is a bit special because publishing is irreversible. My worry is that if we see an X on a CI build we'll think that nothing happened, when it reality perhaps one of the packages failed to publish but the rest were successful.

Reply:
In a monorepo context, this _will_ be run against unpublished packages, so we will encounter this case. We saw this in core. If this exits with a non-zero code, it would show core publish jobs as failing if there was even a single non-published package.